404.html

```diff
@@ -12,6 +12,62 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+
+
+    <!--This block of code is temporary to account for old links since the site was reorganized.
+    This wont redirect _all_ broken links but catches the main ones.-->
+
+    <!--[if lt IE 9]><script type="text/javascript">var IE_fix=true;</script><![endif]-->
+    <script type="text/javascript">
+        
+        window.onload = function ()
+        {
+            var delay = "2000";
+            setTimeout(GoToURL, delay);
+        }
+        function GoToURL()
+        {
+            if(typeof IE_fix != "undefined") // IE8 and lower fix to pass the http referer
+            {
+                var url = "/docs";
+                var referLink = document.createElement("a");
+                referLink.href = url;
+                document.body.appendChild(referLink);
+                referLink.click();
+            }
+            else { 
+                var pathname = window.location.pathname;
+
+                // redirect how-to/build to build
+                if(pathname.includes("/docs/how-to/build")){
+                    window.location.replace("https://onnxruntime.ai/docs/build");
```

Comment:
Can we replace with the relative paths only?

Reply:
We can but then testing doesn't work on the staged sites because github pages has an extra '/onnxruntime/' in the path for staging. I can move back to relative but then you cant test it in the above link.

Reply:
Can we do a more sophisticated string search replace so that the substring of the URL is replaced by the new one, then apply that whole new string to the window.location.replace()? 

With the way it's currently done, it's not going to work on staged sites because it is going to redirect back to the main site anyway

Reply:
updated to relative paths and added a test url in the json redirect file.